### DISCUSSION TITLE
Version policy of bundled Traefik

### DISCUSSION THREAD

**Discussion by ChristianCiach** [k3s-io/k3s#4436] [General]
I often notice that the bundled Traefik uses rather old versions. For example, v1.21.6+k3s1 uses Traefik v2.4.8, even though  v2.4.14 was already available at the time of release. K3s v1.22.3+k3s1 uses Traefik v2.5.0, even though v2.5.3 has long been avaibable.

In my opinion, Traefik is a critical component for security, so it should be updated in a timely manner.

I know that you can disable the bundled Traefik and deploy your own. You could also control the version by setting the `image` using a custom `HelmChartConfig`. But that is not the point. Traefik is bundled for convenience, so I guess most people will just use it as-is. 

So I guess my question is: Why does K3s use outdated versions of Traefik?

**Comment by ChristianCiach** [comment]
Looks like the bundled Traefik gets updated more often now. Thanks!

**Reply by brandond** [reply]
We don't have a hard policy for updating bundled components. We will update when there's a security vulnerability or incompatibility between components. Other than that it's best effort.
